distributed/tests/test_client.py

```diff
@@ -6293,27 +6294,18 @@ async def f():
     assert not s.counters["op"].components[0]["gather"]
 
 
+@pytest.mark.slow
 @gen_cluster(client=True)
 async def test_as_completed_async_for_cancel(c, s, a, b):
     x = c.submit(inc, 1)
-    y = c.submit(sleep, 0.3)
+    y = c.submit(lambda: Event().wait())  # This task will never finish
```

Comment:
The workers typically try to close the threadpool gracefully. If you do not allow the task to finish, that's not possible. I would suggest to let the task finish properly at the end of the test.

Reply:
Done